fenix/app/src/test/java/org/mozilla/fenix/search/SearchDialogControllerTest.kt

```diff
@@ -362,6 +363,116 @@ class SearchDialogControllerTest {
         verify { store.dispatch(SearchFragmentAction.ShowSearchShortcutEnginePicker(false)) }
     }
 
+    @Test
+    fun `WHEN not in private mode THEN do not show prompt`() {
```

Comment:
While I do believe we should add tests for checking the flags inside the [action](https://searchfox.org/mozilla-mobile/source/firefox-android/fenix/app/src/main/java/org/mozilla/fenix/search/SearchDialogController.kt#157), it might be out of scope of this PR. Generally speaking, when adding `val isFeltPrivacyEnabled = settings.feltPrivateBrowsingEnabled`, we would expect the store not to receive any of `AllowSearchSuggestionsInPrivateModePrompt` as the feature should be disabled, right? In the case, capturing the list of events that were dispatched and making sure that there are (or aren't) events of certain type would do the trick. Just wwo tests should be enough.

The next question should be, wouldn't we want to write the tests for other flags in the same manner? It looks like if the suggestions are disabled or we are not in the private mode, no events of `llowSearchSuggestionsInPrivateModePrompt` should be there. It looks like only the tests for empty and not empty text input would actually care about `AllowSearchSuggestionsInPrivateModePrompt` boolean value.

And that would require a bit of refactoring of the logic as well. A possible follow up?👀

Reply:
I was originally going to just write the initial test but I wanted a little more confidence that I wasn't breaking anything else so I wrote a couple more. Ultimately all of these tests will get removed once the feature rolls out as the prompt will no longer exist.